marketplace/src/main/assemble/assembly.xml

```diff
@@ -1,31 +1,190 @@
+           dest="${outdir}/nuxeo-nxr-server-jsf-ui"/>
+    <unzip src="${outdir}/nxr-${maven.project.version}.zip" dest="${outdir}/nuxeo-nxr-server-jsf-ui"/>
+    <zip destfile="${outdir}/nuxeo-nxr-server-jsf-ui-${nuxeo.distribution.version}.zip"
+         basedir="${outdir}/nuxeo-nxr-server-jsf-ui"/>
+    <zip destfile="${outdir}/nxr-${maven.project.version}.zip" basedir="${outdir}/nxr"/>
+
+    <!-- Diff against Nuxeo Server + JSF UI -->
+    <nx:zipdiff file1="${outdir}/nxr-${maven.project.version}.zip"
+                file2="${outdir}/nuxeo-nxr-server-jsf-ui-${nuxeo.distribution.version}.zip"
+                includesfile="${outdir}/includes"
+                excludesfile="${outdir}/excludesfile"
+                patternsetid="shibboleth.versus.jsf">
+    </nx:zipdiff>
+    <unzip src="${outdir}/nxr-${maven.project.version}.zip" dest="${outdir}/marketplace/install">
+      <patternset refid="shibboleth.versus.jsf"/>
+      <firstmatchmapper>
+        <globmapper from="bundles/*" to="bundles-jsf-ui/*"/>
+        <globmapper from="lib/*" to="lib-jsf-ui/*"/>
+      </firstmatchmapper>
+    </unzip>
+    <!-- Generate install.xml content -->
+    <var name="install.bundles" value=""/>
+    <var name="install.bundles"
+         value="${line.separator}
+  &lt;update file=&quot;${package.root}/install/bundles&quot; todir=&quot;${env.bundles}&quot; /&gt;"/>
+    <var name="install.bundles"
+         value="${install.bundles}${line.separator}
+  &lt;update file=&quot;${package.root}/install/bundles-jsf-ui&quot; todir=&quot;${env.bundles}&quot; if=&quot;Packages.contains('nuxeo-jsf-ui')&quot; /&gt;"/>
 
-    <copy todir="${maven.project.build.directory}/marketplace/install/bundles">
-      <artifact:resolveFile key="org.nuxeo.ecm.platform:nuxeo-usermapper::jar" />
+    <var name="install.libs" value=""/>
+    <if>
+      <available file="${outdir}/marketplace/install/lib"/>
+      <then>
+        <var name="install.libs"
+             value="${line.separator}
+  &lt;update file=&quot;${package.root}/install/lib&quot; todir=&quot;${env.lib}&quot; /&gt;"/>
+      </then>
+    </if>
+    <if>
+      <available file="${outdir}/marketplace/install/lib-jsf-ui"/>
+      <then>
+        <var name="install.libs"
+             value="${install.libs}${line.separator}
+  &lt;update file=&quot;${package.root}/install/lib-jsf-ui&quot; todir=&quot;${env.lib}&quot; if=&quot;Packages.contains('nuxeo-jsf-ui')&quot; /&gt;"/>
+      </then>
+    </if>
+
+    <var name="install.properties"
+         value="${line.separator}
+  &lt;copy file=&quot;${package.root}/install/artifacts-shibboleth.properties&quot;${line.separator}
+    todir=&quot;${env.server.home}/sdk&quot; overwrite=&quot;true&quot; /&gt;${line.separator}
+  &lt;copy file=&quot;${package.root}/install/test-artifacts-shibboleth.properties&quot;${line.separator}
+    todir=&quot;${env.server.home}/sdk&quot; overwrite=&quot;true&quot; /&gt;"/>
+
```

Comment:
The Shibboleth MP has a `shibboleth` template that you can find in `src/main/resources/templates/`.
We need to add it to the `install.xml` file so that it will be deployed on the server when installing the Shibboleth MP.

```
  <var name="install.templates" value="" />
    <if>
      <available file="${outdir}/marketplace/install/templates" />
      <then>
        <var name="install.templates"
             value="${line.separator}
  &lt;copy dir=&quot;${package.root}/install/templates&quot; todir=&quot;${env.templates}&quot; overwrite=&quot;true&quot; /&gt;${line.separator}
  &lt;config addtemplate=&quot;shibboleth&quot; /&gt;${line.separator}" />
      </then>
    </if>
```

For instance, the Drive MP does the same thing
https://github.com/nuxeo/marketplace-drive/blob/master/marketplace/src/main/assemble/assembly.xml
https://github.com/nuxeo/marketplace-drive/blob/master/marketplace/src/main/assemble/assembly.xml#L180
https://github.com/nuxeo/marketplace-drive/blob/master/marketplace/src/main/assemble/assembly.xml#L196